translatable_proc/src/data/config.rs

```diff
@@ -116,20 +137,57 @@ static TRANSLATABLE_CONFIG: OnceLock<TranslatableConfig> = OnceLock::new();
 /// - Uses `OnceLock` for thread-safe singleton initialization
 /// - Missing config file is not considered an error
 /// - Config file must be named `translatable.toml` in root directory
+/// - Environment variables take precedence over TOML configuration
+/// - Supported environment variables:
+///   - `LOCALES_PATH`: Overrides translation directory path
+///   - `SEEK_MODE`: Sets file processing order ("alphabetical" or
+///     "unalphabetical")
+///   - `TRANSLATION_OVERLAP`: Sets conflict strategy ("overwrite" or "ignore")
 ///
 /// # Panics
 /// Will not panic but returns ConfigError for:
 /// - Malformed TOML syntax
 /// - Filesystem permission issues
+/// - Invalid environment variable values
 pub fn load_config() -> Result<&'static TranslatableConfig, ConfigError> {
     if let Some(config) = TRANSLATABLE_CONFIG.get() {
         return Ok(config);
     }
 
-    let config: TranslatableConfig =
-        toml_from_str(read_to_string("./translatable.toml")
-            .unwrap_or("".into()) // if no config file is found use defaults.
-            .as_str())?;
+    // Load base configuration from TOML file
+    let toml_content = read_to_string("./translatable.toml").unwrap_or_default();
```

Comment:
As per the other review comment, instead of setters what this could be is first decide what value you are going to have and then construct the instance, so, load all the values from the file, declare possibly Option<> variables that have the ones on the file or none overwrite these variables for the environment variables, and construct an instance with that.unwrap_or(), we could then possibly avoid even using serde for the matter.